src/main/java/hudson/security/LDAPSecurityRealm.java

```diff
@@ -465,54 +472,90 @@ public LDAPSecurityRealm(String server, String rootDN, String userSearchBase, St
         this(server, rootDN, userSearchBase, userSearch, groupSearchBase, groupSearchFilter, groupMembershipStrategy, managerDN, managerPasswordSecret, inhibitInferRootDN, disableMailAddressResolver, cache, environmentProperties, displayNameAttributeName, mailAddressAttributeName, IdStrategy.CASE_INSENSITIVE, IdStrategy.CASE_INSENSITIVE);
     }
 
-    @DataBoundConstructor
+    /**
+     * @deprecated retained for backwards binary compatibility.
+     */
+    @Deprecated
     public LDAPSecurityRealm(String server, String rootDN, String userSearchBase, String userSearch, String groupSearchBase, String groupSearchFilter, LDAPGroupMembershipStrategy groupMembershipStrategy, String managerDN, Secret managerPasswordSecret, boolean inhibitInferRootDN, boolean disableMailAddressResolver, CacheConfiguration cache, EnvironmentProperty[] environmentProperties, String displayNameAttributeName, String mailAddressAttributeName, IdStrategy userIdStrategy, IdStrategy groupIdStrategy) {
-        this.server = server.trim();
-        this.managerDN = fixEmpty(managerDN);
-        this.managerPasswordSecret = managerPasswordSecret;
-        this.inhibitInferRootDN = inhibitInferRootDN;
-        if(!inhibitInferRootDN && fixEmptyAndTrim(rootDN)==null) rootDN= fixNull(inferRootDN(server));
-        this.rootDN = rootDN.trim();
-        this.userSearchBase = fixNull(userSearchBase).trim();
-        userSearch = fixEmptyAndTrim(userSearch);
-        this.userSearch = userSearch!=null ? userSearch : DescriptorImpl.DEFAULT_USER_SEARCH;
-        this.groupSearchBase = fixEmptyAndTrim(groupSearchBase);
-        this.groupSearchFilter = fixEmptyAndTrim(groupSearchFilter);
-        this.groupMembershipStrategy = groupMembershipStrategy == null ? new FromGroupSearchLDAPGroupMembershipStrategy("") : groupMembershipStrategy;
+        this(createLdapConfiguration(server, rootDN, userSearchBase, userSearch, groupSearchBase, groupSearchFilter, groupMembershipStrategy, managerDN, managerPasswordSecret, inhibitInferRootDN, environmentProperties, displayNameAttributeName, mailAddressAttributeName),
+                disableMailAddressResolver, cache, userIdStrategy, groupIdStrategy);
+    }
+
+    @DataBoundConstructor
+    public LDAPSecurityRealm(List<LDAPConfiguration> configurations, boolean disableMailAddressResolver, CacheConfiguration cache, IdStrategy userIdStrategy, IdStrategy groupIdStrategy) {
+        if (configurations == null || configurations.isEmpty()) {
+            //Correct FormException should be handled by DescriptorImpl.newInstance
+            throw new IllegalArgumentException(jenkins.security.plugins.ldap.Messages.LDAPSecurityRealm_AtLeastOne());
+        }
+        this.configurations = configurations;
```

Comment:
because they can do whatever crazy they want in that binding... including defining a custom groovy class that may make assumptions about the classes of the different beans... it's likely a low risk, but it is still a risk... plus nudging them to stop using the custom binding (which they probably do not need any more) is a good thing.

In any case the custom binding will likely not be using any of the settings for the different servers so you will just have an array of identical search beans